crowbar_framework/lib/crowbar/lock.rb

```diff
@@ -0,0 +1,55 @@
+#
+# Copyright 2015, SUSE LINUX GmbH
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+module Crowbar
+  class Lock
+    class Remote
+      def initialize(logger: Rails.logger)
+        @logger = logger
+      end
+
+      def create(nodes, name)
+        nodes.each do |node|
+          @logger.info("Creating remote temporary lockfile #{name} on #{node}")
+          RemoteNode.ssh_cmd(node, "touch /var/lock/#{name}")
+        end
+      end
+
+      def destroy(nodes, name)
+        nodes.each do |node|
+          @logger.info("Destroying remote temporary lockfile #{name} on #{node}")
+          RemoteNode.ssh_cmd(node, "rm -f /var/lock/#{name}")
+        end
+      end
+    end
+
+    class Local
+      def initialize(logger: Rails.logger)
+        @logger = logger
+      end
+
+      def create(name)
+        @logger.info("Creating local temporary lockfile #{name}")
+        File.new("/var/lock/#{name}", "w")
+      end
+
+      def destroy(name)
+        @logger.info("Destroying local temporary lockfile #{name}")
+        File.delete("/var/lock/#{name}")
+      end
+    end
+  end
+end
```

Comment:
not sure if this is the most elegant way with 2 inner classes, please ruby specialists have a look. 
and i still don't like too much the 2 initializers, because they are DRY. can we share the logger somehow among the 2 inner classes? what would be the most elegant way to do that?